beets/importer.py

```diff
@@ -672,7 +673,7 @@ def manipulate_files(self, move=False, copy=False, write=False,
                     # old paths.
                     item.move(copy, link)
 
-            if write and self.apply:
+            if write and (self.apply or self.choice_flag == action.RETAG):
```

Comment:
Or perhaps `self.apply` should be set in the RETAG case? (I'm not certain; it's just worth considering.)